smartsim/experiment.py

```diff
@@ -276,6 +276,82 @@ def get_status(
         stats = (stats_map.get(i, InvalidJobStatus.NEVER_STARTED) for i in ids)
         return tuple(stats)
 
+    def wait(
+        self, *ids: LaunchedJobID, timeout: float | None = None, verbose: bool = True
+    ) -> None:
+        """Block execution until all of the provided launched jobs, represented
+        by an ID, have entered a terminal status.
+
+        :param ids: The ids of the launched ids to wait for.
+        :param timeout: The max time to wait for all of the launched jobs to end.
+        :param verbose: Whether found statuses should be displayed in the console.
+        :raises ValueError: No IDs were provided.
+        """
+        if not ids:
+            raise ValueError("No job ids to wait on provided")
+        self._poll_for_statuses(
+            ids,
+            TERMINAL_STATUSES,
```

Comment:
Just because in theory this could give us a little bit more flexibility

For instance, we might want to add a block to `Experiment.start` want to add a block to make sure that everything started is running a-la:
```py
class Experiment:
    ...
    def start(self, ...):
        ...
        self._poll_for_statuses(ids, {JobStatus.RUNNING}, ...)
        ...
    ...
```
but its all very speculative.